support/on-prem-bootstrap.sh

```diff
@@ -0,0 +1,103 @@
+#!/bin/bash
+
+# The purpose of this script is to download the latest stable version of every
+# package in the core-plans repo, tar them up, and upload to S3. It also supports
+# downloading the archive from S3, extracting it, and uploading to a new depot.
+
+set -e
```

Comment:
Use 'set -euo pipefail' instead?

Reply:
I can add `-o pipefail` but I specifically don't want `-u` because that will cause the graceful error handling below on line 38 to fail.  I'd rather explicitly check for the env vars we expect to be set and gracefully fail, as that's a better UX.

Reply:
Cool, sounds good